### ISSUE TITLE
Failures to update  an installed copy of master branch

### ISSUE THREAD

**Issue by Brian1Gaff** [nvaccess/nvda#7639]
### Steps to reproduce:
Get the  prompt to update
do the required steps. Fails as below

Message presented...
The installation is unable to remove or overwrite a file. Another copy of NVDA may be running on another 

logged-on user account. Please make sure all installed copies of NVDA are shut down and try the 

installation again.

Surely the installer of the update should have done this already?
 The important bits of the log.

Speaking [LangChangeCommand ('en_GB'), u'Updating NVDA  dialog  Please wait while your previous 

installation of NVDA is being updated.']
IO - tones.beep (08:55:50.936):
Beep at pitch 440, for 40 ms, left volume 50, right volume 50
IO - speech.speak (08:55:50.941):
Speaking [LangChangeCommand ('en_GB'), u'Please wait']
IO - tones.beep (08:55:50.973):
Beep at pitch 110.0, for 40 ms, left volume 50, right volume 50
IO - speech.speak (08:55:50.982):
Speaking [LangChangeCommand ('en_GB'), u'0 percent']
ERROR - gui.installerGui.doInstall (08:55:51.605):
Failed to execute installer
Traceback (most recent call last):
  File "gui\installerGui.pyc", line 33, in doInstall
RetriableFailure
IO - tones.beep (08:55:51.611):
Beep at pitch 1760, for 40 ms, left volume 50, right volume 50
DEBUGWARNING - RPC process 3716 (nvda_update_iu3byu.exe) (08:55:58.371):
Thread 3636, build\x86\remote\ia2Support.cpp, IA2Support_inProcess_terminate, 152:
WaitForMultipleObjects returned 258

snip of me doing object nav to read the error again..

IO - speech.speak (08:56:29.878):
Speaking [LangChangeCommand ('en_GB'), u'The installation is unable to remove or overwrite a file. 

Another copy of NVDA may be running on another logged-on user account. Please make sure all installed 

copies of NVDA are shut down and try the installation again. copied to clipboard']
DEBUG - queueHandler.pumpAll (08:56:29.894):
generator 3 finished
IO - inputCore.InputManager.executeGesture (08:56:38.003):
Input: kb(desktop):tab
IO - speech.speak (08:56:38.048):
Speaking [LangChangeCommand ('en_GB'), u'Cancel  button']
IO - inputCore.InputManager.executeGesture (08:56:39.122):
Input: kb(desktop):space
IO - speech._speakSpellingGen (08:56:39.141):
Speaking character u'space'
DEBUG - queueHandler.registerGeneratorObject (08:56:39.141):
Adding generator 4
IO - speech.speak (08:56:39.141):
Speaking [LangChangeCommand ('en_GB'), u'pressed']
DEBUG - queueHandler.pumpAll (08:56:39.154):
generator 4 finished
ERROR - gui.installerGui.doInstall (08:56:39.262):
Installation failed: 
IO - speech.speak (08:56:39.325):
Speaking [LangChangeCommand ('en_GB'), u'Error  dialog  The installation of NVDA failed. Please check the 

Log Viewer for more information.']
IO - speech.speak (08:56:39.335):
Speaking [LangChangeCommand ('en_GB'), u'OK  button']

So at this point we exited, rather brutally!

Next any attempt to run the installed copy came up as cannot find the nvda.exe
 The installer was still kind of running but refused to read the desktop so I had to reboot the machine 

with the autoload of nvda off and manually boot in a portable copy. I did this twice since the first time 

left me with no nvda menus desktop and  context menues working and stuck in my users folders tree. It 

then booted and I started the portable copy and reran the installer which I had saved as a portable 

version to a dummy folder and could then use to create the  installed version. this time it worked of 

course as no files in the install folder were in use or locked  which seems to be the main issue when 

this failure occurs. There needs to be some kind of order change in file copying so any issue like this 

does not result in a non working copy of nvda part way through. as to why its happening when it never 

ever used to till recently. I'll leve to the experts!
> A list of the steps you take to demonstrate the problem. 
See above, its easier this way!>
### Expected behavior:
It should proceed with a few seconds of bleeping then reboot the new copy> 
### Actual behavior:
It gives up with an error as in the main fault description above.

### System configuration:
NVDA version:

Cannot tell you which version but it was two older than the latest one to start with.
Now its fixed its...NVDA version master-14476,a826028e

NVDA Installed or portable:
Installed master branch
Other information:

Add ons present in the installed version at the time.
clipspeak, pico, speech palyer in espeak, golden cursor focus highliigh resource monitor extended winamp  dectalk synth goldwave 
All up to date, active synth Espeak with quincy voice.

Windows version:
Windows 7 pro 64 and sp1, completely up to date.

Does the issue still occur after restarting your PC?

Well seeing as its very flakey and actually deletes bits of nvda, obviously not but of course its due to a file in use issue, so eventually no it can be fixed by an install if a different version of nvda launched the installer.
